frontend/packages/core/src/Table/table.tsx

```diff
@@ -4,65 +4,188 @@ import {
   IconButton,
   Paper as MuiPaper,
   Table as MuiTable,
-  TableBody,
+  TableBody as MuiTableBody,
   TableCell as MuiTableCell,
   TableContainer as MuiTableContainer,
-  TableHead,
+  TableHead as MuiTableHead,
   TableProps as MuiTableProps,
   TableRow as MuiTableRow,
   TableRowProps as MuiTableRowProps,
+  useMediaQuery,
 } from "@material-ui/core";
 import MoreVertIcon from "@material-ui/icons/MoreVert";
 
 import { Popper, PopperItem } from "../popper";
+import { Typography } from "../typography";
 
-const TablePaper = styled(MuiPaper)({
+const StyledPaper = styled(MuiPaper)({
   border: "1px solid #E7E7EA",
 });
 
-const StyledTableRow = styled(MuiTableRow)({
-  ":hover": {
-    background: "#EBEDFB",
+const StyledTable = styled(MuiTable)<{
+  actions: boolean;
+  columnCount: number;
+  compress: boolean;
+  responsive?: boolean;
+}>(
+  {
+    minWidth: "100%",
+    borderCollapse: "collapse",
+    alignItems: "center",
   },
+  props => ({
+    display: !props.responsive ? "table" : props.compress ? "table" : "grid",
+    gridTemplateColumns: `repeat(${props.columnCount}, auto)${props.actions ? " 80px" : ""}`,
+  })
+);
+
+const StyledTableBody = styled(MuiTableBody)({
+  display: "contents",
 });
 
-export const TableCell = styled(MuiTableCell)(
+const StyledTableHead = styled(MuiTableHead)({
+  display: "contents",
+  backgroundColor: "#D7DAF6",
+});
+
+const StyledTableRow = styled(MuiTableRow)<{ responsive?: boolean }>(
   {
+    ":nth-child(even)": {
+      background: "rgba(13, 16, 48, 0.03)",
```

Comment:
<img width="600" alt="Screen Shot 2021-08-05 at 12 11 34 PM" src="https://user-images.githubusercontent.com/39421794/128384111-8d38eb8e-d875-47a5-8e1b-f9d1d3777d20.png">

do you think this should maybe be prop based? in case users don't want a diff color for every other row

Reply:
Maybe, this is just going off of the designs. I chose a random header color as well so we can revisit this when Kanchan comes back with finalized designs